### PR TITLE
Adding ManagedResourceGroupSettings to RegistryRP in 2024-07-01-Preview

### PR DESCRIPTION
**Pull request by theodoreschen**
# Choose a PR Template

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

### CODE REVIEW

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-07-01-preview/registries.json [hunk 1]
```diff
@@ -710,6 +715,34 @@
...
+      },
+      "additionalProperties": false
+    },
+    "ManagedResourceGroupAssignedIdentities": {
+      "description": "Details for managed resource group assigned identities.",
+      "type": "object",
+      "properties": {
+        "principalId": {
+          "description": "Identity principal Id",
+          "type": "string",
```
Comment (madhura-np) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-07-01-preview/registries.json:740]: >type [](http://example.com/codeflow?start=11&length=4)

"format":"uuid"

File: specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-07-01-preview/registries.json [hunk 2]
```diff
@@ -710,6 +715,34 @@
...
+          "x-ms-identifiers": []
+        }
+      },
+      "additionalProperties": false
+    },
+    "ManagedResourceGroupAssignedIdentities": {
+      "description": "Details for managed resource group assigned identities.",
+      "type": "object",
+      "properties": {
+        "principalId": {
```
Comment (madhura-np) [inline specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-07-01-preview/registries.json:738]: >principalId [](http://example.com/codeflow?start=9&length=11)

should this be made "readOnly": true

**Comment by raosuhas** [comment]
AS discussed in office hours this is an acceptable use of uuid. Please add a suppression for this and restart ARM review by removing the tags

**Comment by theodoreschen** [comment]
> AS discussed in office hours this is an acceptable use of uuid. Please add a suppression for this and restart ARM review by removing the tags

it looks like that's no longer being checked and is no longer a blocker? Do I still need to add a suppression then?

**Comment by theodoreschen** [comment]
> suppressions

is there an example of how to suppress a LintDiff error?

**Comment by theodoreschen** [comment]
> Could you please add a suppressions for the failing Lint Validation ? Is a valid usage of the UUID since it is the AAD objectid

Is there an example of how to suppress Lint Validations? The docs only seem to explain how to suppress SDK breaking changes

**Comment by raosuhas** [comment]
> > Could you please add a suppressions for the failing Lint Validation ? Is a valid usage of the UUID since it is the AAD objectid
> 
> Is there an example of how to suppress Lint Validations? The docs only seem to explain how to suppress SDK breaking changes

This is all explained if you used the correct template for the control plane API. For future reference , please do not skip that step above. 

Pasting the relevant section here for now : 

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

**Review by ZhidaLiu — APPROVED** [review]

**Pull request merged**